elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCleanerThread.java

```diff
@@ -48,7 +52,13 @@ protected void deleteAsyncQuery() {
 
         String filterExpression = "createdOn=le='" + cleanupDateFormatted + "'";
 
-        asyncQueryDao.deleteAsyncQueryAndResultCollection(filterExpression);
+        Collection<AsyncQuery> asyncQueryList = asyncQueryDao.deleteAsyncQueryAndResultCollection(filterExpression);
+        Iterator itr = asyncQueryList.iterator();
+
+        while (itr.hasNext()) {
+            AsyncQuery query = (AsyncQuery) itr.next();
+            resultStorageEngine.deleteResultsByID(query.getId());
```

Comment:
I think its better to rename the function deleteResultsByID to deleteResultsCollection. It should take `Collection<AsyncQuery>` as an argument.  All of the iterator logic can be then moved to the function implementation. The reason being in our database interpretation we have to delete each row one by one but for a filesystem-based, we can delete multiple at the same time.